### PR TITLE
File preview into bugbash

### PR DESCRIPTION
**Pull request by nathandtam**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Implements file preview/download with images, pdfs, and apks.
- Build and style the files page (`.../[lessonId]/files`).
- Adds a table (`FilesTable`), held in the files page, that holds and can sort the files.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

**new page styling:**
Screenshot 2025-12-09 at 11 23 42 PM

**functionality walkthrough:**

https://github.com/user-attachments/assets/0dc6fb5e-406a-4fb4-9172-0e79f74ec4a2

## 👍 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## 👍 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
**What's NOT in this PR:**
- files do not have date added or date modified property yet, they just use placeholders.
- Some components, like `FileTypeDropdown` aren't functional.
- The interaction for downloading files is not yet styled. Shown here:
Screenshot 2025-12-09 at 11 39 45 PM

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @nehaahussain

### CODE REVIEW

**Review by nehaahussain — CHANGES_REQUESTED** [review]

File: app/[lessonName]/files/page.tsx [hunk 1]
```diff
@@ -1,32 +1,176 @@
...
-            <p>Date Modified</p>
-            <p>File Size</p>
-            <p>File Type</p>
-          </OtherHeaders>
-        </FileHeaders>
-      </FileContainer>
+      <LessonHeader>
+        <BackButton>
+          <BackButtonIconSlot>
+            <svg
```
Comment (nehaahussain) [inline app/[lessonName]/files/page.tsx]: can we move all svgs into `lib/icons.tsx`? see #22

File: app/[lessonName]/files/page.tsx [hunk 2]
```diff
@@ -1,32 +1,176 @@
...
+          <span>My Lessons</span>
+        </BackButton>
+      </LessonHeader>
+
+      <HeaderRow>
+        <Title>{lessonName}</Title>
+
+        <HeaderRight>
+          {/* Search bar (non-functional, just UI) */}
+          <SearchBar>
```
Comment (nehaahussain) [inline app/[lessonName]/files/page.tsx]: can you move all search bar related code to another component? we have a search bar component already so maybe you can refactor it to take in props and be usable for both lessons page and this files page. or just make a new one for files — up to you

File: app/[lessonName]/files/page.tsx [hunk 3]
```diff
@@ -1,32 +1,80 @@
...
+}) {
+  return (
+    <ModalBackdrop onClick={onClose}>
+      <ModalPanel onClick={e => e.stopPropagation()}>
+        <ModalContent>{children}</ModalContent>
+      </ModalPanel>
+    </ModalBackdrop>
+  );
+}
+
```
[the remark this answers is not in the document]
Comment (nathandtam) [reply app/[lessonName]/files/page.tsx]: moved, renamed to `FilePreviewModal`.

File: app/[lessonName]/files/page.tsx [hunk 4]
```diff
@@ -1,32 +1,80 @@
...
 export default function FilesPage() {
   const lessonName = useParams().lessonName;
+  const [files, setFiles] = useState<LocalFile[]>([]);
+  const [selectedFile, setSelectedFile] = useState<LocalFile | null>(null);
+
+  useEffect(() => {
+    async function load() {
+      const localData = await fetchLocalDatabase();
```
[the remark this answers is not in the document]
Comment (nathandtam) [reply app/[lessonName]/files/page.tsx]: done
